File: LabSoapJava/src/main/java/pl/kamilszustak/labsoapjava/client/SoapConnector.java
package pl.kamilszustak.labsoapjava.client;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.ws.client.core.support.WebServiceGatewaySupport;

public class SoapConnector extends WebServiceGatewaySupport {
    @Value("${base-url}")
    private String baseUrl;

    public Object call(Object request) {
        return getWebServiceTemplate().marshalSendAndReceive(baseUrl, request);
    }
}

File: LabSoap/src/main/kotlin/pl/kamilszustak/labsoap/client/SoapConnector.kt
package pl.kamilszustak.labsoap.client

import org.springframework.beans.factory.annotation.Value
import org.springframework.ws.client.core.support.WebServiceGatewaySupport

class SoapConnector : WebServiceGatewaySupport() {
    @Value("\${base-url}")
    private lateinit var baseUrl: String

    fun callService(request: Any): Any? =
        webServiceTemplate.marshalSendAndReceive(baseUrl, request)
}